Add support for Xiaomi Redmi Note 5 (whyred)

minlexx opened the pull request:
Initial feature set includes:
 * framebuffer display
 * regulators
 * volume/power keys
 * gpio hall sensor
 * battery fulegauge & charger
 * PMIC vibrator
 * PMIC WLED backlight
 * notification LED
 * storage: eMMC, SD cards
 * USB (only USB2 and only gadget mode, like the rest of the platform)
 * hardware video enocder/decoder (venus)
 * WCN3990 Bluetooth, Wi-Fi
 * Synaptics rmi4 touchscreen
 * Display (mdss), Adreno 509 (partial)
 * LCD IPS video mode panel (Tianma TD4310)
 * sensors (accelerometer) are automagically supported too

This one passes dt-validate step much better.

MarvelMathesh committed:
arm64: dts: qcom: add device-tree for Redmi Note 5 Pro (whyred)

Add initial support for Xiaomi Redmi Note 5 Pro smarthone, based on
Snapdragon 636 SoC.

Initial feature set includes:

 * framebuffer display
 * regulators
 * volume/power keys
 * gpio hall sensor
 * battery fulegauge & charger
 * PMIC vibrator
 * PMIC WLED backlight
 * notification LED (LPG)
 * storage: eMMC, SD cards
 * USB (only USB2 and only gadget mode, like the rest of the platform)
 * hardware video enocder/decoder (venus)
 * Bluetooth, Wi-Fi (WCN3990)
 * Synaptics E753 RMI4 touchscreen
 * Display (mdss), Adreno 509 (partial)
 * LCD IPS video mode panel (Tianma TD4310)

Signed-off-by: Mathesh Velayudan <123v.mathesh@gmail.com>
Signed-off-by: Alexey Minnekhanov <alexeymin@postmarketos.org>

minlexx committed:
dt-bindings: arm: qcom: Add Xiaomi Redmi Note 5 Pro

Xiaomi Redmi Note 5 Pro (xiaomi-whyred) is a midrange priced
smartphone from 2018 and is based on the SDM636 SoC.

Signed-off-by: Alexey Minnekhanov <alexeymin@postmarketos.org>

M0Rf30 commented on arch/arm64/boot/dts/qcom/sdm636-xiaomi-whyred.dts:
```diff
@@ -0,0 +1,547 @@
+	status = "okay";
+};
+
+&pm660_fg {
```
```
&pm660_fg {
        monitored-battery = <&battery>;
        power-supplies = <&pm660_charger>;

        status = "okay";
};
```

minlexx replied:
riiight

M0Rf30 commented on arch/arm64/boot/dts/qcom/sdm636-xiaomi-whyred.dts:
```diff
@@ -0,0 +1,547 @@
+	vdd-1.8-xo-supply = <&vreg_l9a_1p8>;
+	vdd-1.3-rfa-supply = <&vreg_l6a_1p3>;
+	vdd-3.3-ch0-supply = <&vreg_l19a_3p3>;
+	vdd-3.3-ch1-supply = <&vreg_l19a_3p3>;
```
Just as reminder
`vdd-3.3-ch1-supply = <&vreg_l19a_3p3>;`
 is only present in tulip to remove a dmesg message, related to missing supply
 Maybe should be applied also to other devices?

minlexx replied:
ah, ch1-supply is optional, right? perhaps it can be removed then, and don't care about dmesg message..

M0Rf30 replied:
Or is it right to assume that two channels are served by the same vreg?and explictly add both? from a functional perspective seems that nothing change,

minlexx replied:
I have no idea about this one. Example in dt-bindings has ch0 and ch1 on 2 different regulators, but only 2 properties are marked as required: compatible and reg...

barni2000 commented on arch/arm64/boot/dts/qcom/sdm636-xiaomi-whyred.dts:
```diff
@@ -0,0 +1,606 @@
+		compatible = "gpio-keys";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_hall_sensor_default>;
```
```suggestion
		pinctrl-names = "default";
```

barni2000 commented on arch/arm64/boot/dts/qcom/sdm636-xiaomi-whyred.dts:
```diff
@@ -0,0 +1,606 @@
+	gpio-hall-sensor {
+		compatible = "gpio-keys";
+
+		pinctrl-names = "default";
```
```suggestion
		pinctrl-0 = <&gpio_hall_sensor_default>;
```

barni2000 commented on arch/arm64/boot/dts/qcom/sdm636-xiaomi-whyred.dts:
```diff
@@ -0,0 +1,606 @@
+	status = "okay";
+
+	zap-shader {
+		firmware-name = "a512_zap.mbn";
```
```suggestion
		firmware-name = "qcom/sdm660/whyred/a512_zap.mbn";
```
I would recommend some prefix for avoid conflicts later.

minlexx replied:
For this change it's already too late, all our other DTs use no path.

If this needs to be changed one day, it should be done for all of them in mass in some other MR..

minlexx replied:
I'll consider this for later switch to generidc device (if it ever happens)

barni2000 commented on arch/arm64/boot/dts/qcom/sdm636-xiaomi-whyred.dts:
```diff
@@ -0,0 +1,606 @@
+		#size-cells = <0>;
+		interrupts-extended = <&tlmm 67 IRQ_TYPE_EDGE_FALLING>;
+
+		pinctrl-names = "default", "sleep";
```
pinctrl-names should come after pinctrl-1

minlexx replied:
it should indeed

minlexx committed:
dt-bindings: display: panel: Add Tianma TD4310 panel

Document Tianma TD4310 panel used in Xiaomi Redmi Note 5 Pro smartphone.
It is an IPS 1080x2160 DSI video mode panel.

Signed-off-by: Alexey Minnekhanov <alexeymin@postmarketos.org>

minlexx committed:
drivers/gpu: panel: Add support for Tianma TD4310 panel

Add driver for IPS 1080x2160 DSI video mode panel used in
Xiaomi Redmi Note 5 Pro (codename whyred) smartphone.

Signed-off-by: Alexey Minnekhanov <alexeymin@postmarketos.org>

minlexx committed:
fixup! sdm660_defconfig

minlexx commented:
I think for now I've done everything I could, this now includes support for display/GPU/panel too.

![photo_2025-03-31_03-37-17](https://github.com/user-attachments/assets/d1d9d654-7786-4cb1-89ab-657efdb7daa4) 
![photo_2025-03-31_03-45-56](https://github.com/user-attachments/assets/81c59677-e365-483b-885e-2a954a402dce)
![photo_2025-03-31_14-59-03](https://github.com/user-attachments/assets/5835743a-8bfa-40bc-8de1-aadb32fe115b)
